lib/packages.cf

```diff
@@ -2255,3 +2255,21 @@ bundle agent package_specific(package_name, desired, package_version, package_ar
     "(DEBUG|DEBUG_$(this.bundle)).filebased.!(suse|sles).!debian.!redhat.!aix.!solaris_pkgadd"::
       "DEBUG $(this.bundle): sorry, can't do file-based installs on $(sys.os)";
 }
+
+bundle agent render_package_modules
+{
+  vars:
+    "template_dir" string => "$(this.promise_dirname)/templates/packages$(const.dirsep)";
+    "mustaches" slist => findfiles("$(template_dir)*.mustache");
+    "modules" slist => maplist(regex_replace("$(this)", "$(template_dir)(.*).mustache", "$1", "g"), @(mustaches));
+
+  files:
+    "$(sys.workdir)/modules/packages/$(modules)"
```

Comment:
@nickanderson is there a simple way to not render this file if the newly rendered file is different than the old one?

aka, only write the file if there is nothing there already, preserve user customizations.

I'm sure it can be done with a healthy chunk of policy, just wondering if some odd attribute is available.

Reply:
I suppose we could find files in the target and add classes for them. :)

Reply:
and if already there and the file is different then spit out a not kept promise of some sort, but diffing would require rendering to a temp dir I think, right?